### PR TITLE
Implements Changeling Headcrabs

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Implements TG's changeling headcrabs.

This is a "last resort" ability that gibs the user and puts the person in control of a small headcrab---the headcrab can then walk around and either attack things (for a paltry 5 damage) or (and the real purpose), seek out a corpse to inject eggs into.

If eggs are injected into a corpse the headcrab will then die 10 seconds later. The egg that's inside the corpse will, after 4 minutes (unless it's removed), hatch into a monkey with the original changeling's brain, eyes, and mind.  After hatching, it will immediately gib the corpse and burst from it. After this, the changeling can continue its nefarious deeds and abilities.

Related
- Fixes carbons not throwing their organs on gibbing
- Ensures that humans eject mobs in their stomach when gibbed
- Uses forcemove to set location for ejected stomach mobs

:cl: Fox McCloud
rscadd: Implements Changeling headcrabs/headspiders
bugfix: Fixes xenos not throwing their organs when gibbed
bugfix: Fixes swallowed mobs not being ejected when a human is gibbed
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
Do changelings keep all the DNA they've collected if they become monkey?

**Comment by Fox-McCloud** [comment]
>Do changelings keep all the DNA they've collected if they become monkey?

I believe so, yes; they definitely keep their original form.

**Comment by DaveTheHeadcrab** [comment]
Headcrab approved.

**Comment by Bigtimetripod** [comment]
Time to start hiding bodies in maint. Quick question, does it work on husks?

**Comment by Fox-McCloud** [comment]
>Time to start hiding bodies in maint. Quick question, does it work on husks?  It works on any dead carbon---yes, that means even dead xeno bodies.

**Comment by theColdflame** [comment]
do lings need yet another get out of jail free card?

**Comment by FlattestGuitar** [comment]
I can't tell whether this is a standard ability or something you have to evolve, but it does seem very powerful, especially considering lings can already do the monkey transformation.

**Comment by Fox-McCloud** [comment]
@FlattestGuitar 

You have to evolve it.

**Comment by FlattestGuitar** [comment]
I suppose it's a fair trade off then. :+1:

**Comment by Bigtimetripod** [comment]
2 points or even 3 go under it?

File: code/game/gamemodes/changeling/powers/headcrab.dm [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	explosion(get_turf(user),0,0,2,0,silent=1)
+	for(var/mob/living/carbon/human/H in range(2,user))
+		H << "<span class='userdanger'>You are blinded by a shower of blood!</span>"
+		H.Stun(1)
+		H.eye_blurry = max(20, H.eye_blurry)
+		var/obj/item/organ/internal/eyes/E = H.get_int_organ(/obj/item/organ/internal/eyes)
+		if(istype(E))
+			E.damage = max(E.damage+5, 0)
+		H.confused += 3
+	for(var/mob/living/silicon/S in range(2,user))
```
Comment (tigercat2000) [inline code/game/gamemodes/changeling/powers/headcrab.dm]: space above please

File: code/game/gamemodes/changeling/powers/headcrab.dm [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		S << "<span class='userdanger'>Your sensors are disabled by a shower of blood!</span>"
+		S.Weaken(3)
+	var/turf = get_turf(user)
```
Comment (tigercat2000) [inline code/game/gamemodes/changeling/powers/headcrab.dm]: space above please

File: code/game/gamemodes/changeling/powers/headcrab.dm [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	spawn(5) // So it's not killed in explosion
+		var/mob/living/simple_animal/hostile/headcrab/crab = new(turf)
+		for(var/obj/item/organ/internal/I in organs)
+			I.loc = crab
+		crab.origin = M
+		if(crab.origin)
+			crab.origin.active = 1
+			crab.origin.transfer_to(crab)
+			crab << "<span class='warning'>You burst out of the remains of your former body in a shower of gore!</span>"
+	user.gib()
```
Comment (tigercat2000) [inline code/game/gamemodes/changeling/powers/headcrab.dm]: space above please

File: code/modules/mob/living/carbon/carbon.dm [hunk 4]
```diff
@@ -93,13 +93,18 @@
 #undef STOMACH_ATTACK_DELAY
 
 /mob/living/carbon/gib()
+	for(var/obj/item/organ/internal/I in internal_organs)
+		if(istype(loc,/turf))
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/carbon.dm]: isturf loc

**Pull request merged**